src/cli/grain.js

```diff
@@ -73,6 +78,64 @@ const grainCommand: Command = async (args, std) => {
   return 0;
 };
 
+/**
+ * Merge receipts under the same name into one.
+ */
+function mergeReceipts(
+  distributions: $ReadOnlyArray<Distribution>
+): $ReadOnlyArray<GrainReceipt> {
+  const mergedBalances: {[string]: G.Grain} = {};
+  for (const {allocations} of distributions) {
+    for (const {receipts} of allocations) {
+      for (const {amount, id} of receipts) {
+        if (!mergedBalances[id.toString()]) {
+          mergedBalances[id.toString()] = amount;
+        } else {
+          const existingBalance = mergedBalances[id.toString()];
+          mergedBalances[id.toString()] = G.add(amount, existingBalance);
+        }
+      }
+    }
+  }
+
+  return Object.keys(mergedBalances).map((id) => ({
+    amount: mergedBalances[id],
+    id: uuidFromString(id),
+  }));
+}
+
+function printAllocationMarkdownTable(
```

Comment:
I think we can get much cleaner type signatures. I'm thinking stuff like:

```
fn allocationMarkdownSummary(GrainAllocation, Ledger): string;
// maybe calls the above to print a summary for each allocation
fn distributionMarkdownSummary(GrainDistribution, Ledger): string;
```

These helpers probably shouldn't live inside the `cli` module, since they are of general utility. E.g. I can easily imagine wanting to depend on them inside observable. Then the `cli/grain.js` file just imports those and logs the resultant tables to console.

Reply:
Agreed, I plus we can abstract these summary utils to output data of any type, which will be very helpful down the road.